diff --git a/.depend b/.depend
index 2debc6f..49b98f2 100644
--- a/.depend
+++ b/.depend
@@ -4635,6 +4635,7 @@ lambda/matching.cmo : \
     typing/env.cmi \
     lambda/debuginfo.cmi \
     typing/data_types.cmi \
+    typing/ctype.cmi \
     utils/clflags.cmi \
     parsing/asttypes.cmi \
     lambda/matching.cmi
@@ -4659,6 +4660,7 @@ lambda/matching.cmx : \
     typing/env.cmx \
     lambda/debuginfo.cmx \
     typing/data_types.cmx \
+    typing/ctype.cmx \
     utils/clflags.cmx \
     parsing/asttypes.cmx \
     lambda/matching.cmi
diff --git a/lambda/matching.ml b/lambda/matching.ml
index 6079b71..aa0c975 100644
--- a/lambda/matching.ml
+++ b/lambda/matching.ml
@@ -2331,7 +2331,7 @@ let get_expr_args_record ~scopes head { arg; mut; _ } rem =
       rem
     else
       let lbl = all_labels.(pos) in
-      let ptr = Typeopt.maybe_pointer_type head.pat_env lbl.lbl_arg in
+      let ptr = Typeopt.maybe_pointer_type (Env.remove_local_equations head.pat_env) lbl.lbl_arg in
       let access =
         match lbl.lbl_repres with
         | Record_regular
diff --git a/typing/env.ml b/typing/env.ml
index 7de0520..5b48850 100644
--- a/typing/env.ml
+++ b/typing/env.ml
@@ -2384,6 +2384,8 @@ type local_equations = Types.type_declaration Path.Map.t
 let freeze_local_equations env = env.local_constraints
 let restrict_local_equations local_constraints env =
   { env with local_constraints }
+let remove_local_equations env =
+  { env with local_constraints = Path.Map.empty }
 
 (* Non-lazy version of scrape_alias *)
 let scrape_alias t mty =
diff --git a/typing/env.mli b/typing/env.mli
index d5ae2ea..b6e4f43 100644
--- a/typing/env.mli
+++ b/typing/env.mli
@@ -342,6 +342,7 @@ val add_local_constraint: Path.t -> type_declaration -> t -> t
 type local_equations
 val freeze_local_equations: t -> local_equations
 val restrict_local_equations: local_equations -> t -> t
+val remove_local_equations: t -> t
 
 (* Insertion of persistent signatures *)
 
