diff --git a/networkx/algorithms/centrality/group.py b/networkx/algorithms/centrality/group.py
index c78bce04..9826105d 100644
--- a/networkx/algorithms/centrality/group.py
+++ b/networkx/algorithms/centrality/group.py
@@ -43,12 +43,16 @@ def group_betweenness_centrality(G, C, normalized=True, weight=None, endpoints=F
       A NetworkX graph.
 
     C : list or set or list of lists or list of sets
-      A group or a list of groups containing nodes which belong to G, for which group betweenness
-      centrality is to be calculated.
+      A group or a list of groups containing nodes which belong to G,
+      for which group betweenness centrality is to be calculated.
 
     normalized : bool, optional (default=True)
-      If True, group betweenness is normalized by `1/((|V|-|C|)(|V|-|C|-1))`
-      where `|V|` is the number of nodes in G and `|C|` is the number of nodes in C.
+      If True, group betweenness is normalized by $1/(N_{out}(N_{out}-1))$
+      where $N_{out}$ is the number of nodes in `G` that are not in `C`.
+      This ensures the reported value is between 0 and 1.
+      If `endpoints` is True, the normalization uses all nodes in `G`.
+      The reported value is then between $2N_{in}/(N-1)$ and 1 where $N_{in}$
+      is the number of nodes in `C` and $N$ the number of nodes in `G`.
 
     weight : None or string, optional (default=None)
       If None, all edge weights are considered equal.
@@ -56,17 +60,32 @@ def group_betweenness_centrality(G, C, normalized=True, weight=None, endpoints=F
       The weight of an edge is treated as the length or distance between the two sides.
 
     endpoints : bool, optional (default=False)
-      If True include the endpoints in the shortest path counts.
+      By default, only node-pairs that are both not in `C` are counted for
+      group betweenness centrality. The count is how many non-`C` node-pairs have
+      nodes from `C` "between" them on a shortest path.
+
+      When ``endpoints=True``, we also count node-pairs with one or both nodes
+      in `C` while considering endpoint nodes as being between the node-pairs.
+      So we count paths that start in `C` whether or not they pass through
+      any other nodes in `C`. This adds centrality to large groups without any
+      reference to the connectivity of the group. The minimum normalized score
+      is $N_{in}(N_{in}-1)/(N(N-1))$ instead of 0. For that reason, this feature
+      is rarely used.
+
+      We don't currently support considering node-pairs with nodes in `C` without
+      also counting their endpoints. Nor do we support counting endpoints while
+      only considering node-pairs that are both not in `C`. This keyword indicates
+      both coutning endpoints of paths and allowing node-pairs in C.
 
     Raises
     ------
     NodeNotFound
-       If node(s) in C are not present in G.
+       If node(s) in `C` are not present in `G`.
 
     Returns
     -------
     betweenness : list of floats or float
-       If C is a single group then return a float. If C is a list with
+       If `C` is a single group then return a float. If `C` is a list with
        several groups then return a list of group betweenness centralities.
 
     See Also
@@ -75,11 +94,10 @@ def group_betweenness_centrality(G, C, normalized=True, weight=None, endpoints=F
 
     Notes
     -----
-    Group betweenness centrality is described in [1]_ and its importance discussed in [3]_.
-    The initial implementation of the algorithm is mentioned in [2]_. This function uses
-    an improved algorithm presented in [4]_.
+    Group betweenness centrality is defined in [1]_ and discussed in [3]_.
+    The algorithm is described in [2]_ and is based on techniques mentioned in [4]_.
 
-    The number of nodes in the group must be a maximum of n - 2 where `n`
+    The number of nodes in the group must be a maximum of ``N - 2`` where ``N``
     is the total number of nodes in the graph.
 
     For weighted graphs the edge weights must be greater than zero.
@@ -124,6 +142,10 @@ def group_betweenness_centrality(G, C, normalized=True, weight=None, endpoints=F
     if set_v - G.nodes:  # element(s) of C not in G
         raise nx.NodeNotFound(f"The node(s) {set_v - G.nodes} are in C but not in G.")
 
+    # pre-process connectedness for no endpoints treatment
+    is_directed = G.is_directed()
+    connected = nx.is_strongly_connected(G) if is_directed else nx.is_connected(G)
+
     # pre-process dict-of-dicts: path btwn(PB), short path counts(sigma), distances(D)
     PB, sigma, D = _group_preprocessing(G, set_v, weight)
 
@@ -180,20 +202,24 @@ def group_betweenness_centrality(G, C, normalized=True, weight=None, endpoints=F
                             # update sig_xv for future y-values
                             sig_xv -= sig_xvy
         # endpoints
-        N, c = len(G), len(group)
-        if not endpoints:
+        N = len(G)
+        if endpoints:
+            Nscale = N
+        else:  # No endpoints -- usual case
+            N_in = len(group)
+            N_out = Nscale = N - N_in
             # if the graph is connected then subtract the endpoints from
             # the count for all the nodes in the graph. else count how many
             # nodes are connected to the group's nodes and subtract that.
-            if nx.is_directed(G):
-                if nx.is_strongly_connected(G):
-                    extra = c * (2 * N - c - 1)
-                else:
-                    # count paths for group to or from anything
-                    reachables = ((u, v) for u in G for v in D[u] if v != u)
-                    extra = sum(1 for u, v in reachables if (u in group or v in group))
-            elif nx.is_connected(G):
-                extra = c * (2 * N - c - 1)
+            if connected:
+                # N_in * (N_out + N_in-1 + N_out) : in*out + in*(in-1) + out*in
+                # same as N_in * (N-1 + N_out) : in*(all-1) + out*in
+                # paper uses equiv: N_in * (2 * N - N_in - 1)
+                extra = N_in * (N - 1 + N_out)
+            elif is_directed:
+                # count paths for group to or from anything
+                reachables = ((u, v) for u in G for v in D[u] if v != u)
+                extra = sum(1 for u, v in reachables if (u in group or v in group))
             else:
                 # count paths for group to anything
                 # (use 2 if v not in group to shorten reachables)
@@ -202,14 +228,10 @@ def group_betweenness_centrality(G, C, normalized=True, weight=None, endpoints=F
 
             GBC_group -= extra
 
-        # normalized
+        # scale down for normalized or by 2 for undirected
         if normalized:
-            N_out = N - c
-            scale = 1 / (N_out * (N_out - 1))
-            GBC_group *= scale
-
-        # If undirected than count only the undirected edges
-        elif not G.is_directed():
+            GBC_group /= Nscale * (Nscale - 1)
+        elif not is_directed:
             GBC_group /= 2
 
         GBC.append(GBC_group)
@@ -325,10 +347,10 @@ def prominent_group(
 
     Notes
     -----
-    Group betweenness centrality is described in [1]_ and its importance discussed in [3]_.
+    Group betweenness centrality is defined in [1]_ and discussed in [3]_.
     The algorithm is described in [2]_ and is based on techniques mentioned in [4]_.
 
-    The number of nodes in the group must be a maximum of ``n - 2`` where ``n``
+    The number of nodes in the group must be a maximum of ``N - 2`` where ``N``
     is the total number of nodes in the graph.
 
     For weighted graphs the edge weights must be greater than zero.
@@ -620,7 +642,7 @@ def group_closeness_centrality(G, S, weight=None):
     It is assumed that 1 / 0 is 0 (required in the case of directed graphs,
     or when a shortest path length is 0).
 
-    The number of nodes in the group must be a maximum of n - 1 where `n`
+    The number of nodes in the group must be a maximum of ``N - 1`` where ``N``
     is the total number of nodes in the graph.
 
     For directed graphs, the incoming distance is utilized here. To use the
