diff --git a/networkx/algorithms/centrality/group.py b/networkx/algorithms/centrality/group.py
index 9826105d..bb058537 100644
--- a/networkx/algorithms/centrality/group.py
+++ b/networkx/algorithms/centrality/group.py
@@ -451,8 +451,7 @@ def prominent_group(
     # If undirected then count only the undirected edges
     elif not G.is_directed():
         max_GBC /= 2
-    max_GBC = float(f"{max_GBC:.2f}")
-    return max_GBC, max_group
+    return max_GBC.item(), max_group
 
 
 def _dfbnb(G, k, DF_tree, max_GBC, root, D, max_group, nodes, greedy):
